pennylane/grouping/pauli.py

```diff
@@ -250,3 +253,80 @@ def pauli_mult_with_phase(pauli_1, pauli_2, wire_map=None):
             phase *= -1j
 
     return pauli_product, phase
+
+
+@lru_cache()
```

Comment:
how did you know to use this decorator here? I don't really understand how caching works and when to take advantage of it!

Reply:
There's probably better explanations out there, but from my perspective `partition_pauli_group` fits the criteria nicely of:
- simple inputs that can be hashed
- function takes a long time to compute
- restricted set of inputs that may be called multiple times (e.g., in circuit cutting we might have a few fragments with 3-wire measure nodes)